demo/react/src/VoiceWidget.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useCallback, useEffect, useState, useRef } from "react";
+import { useCheetah } from "@picovoice/cheetah-react";
+
+import cheetahModel from "./lib/cheetahModel";
+
+export default function VoiceWidget() {
+  const accessKeyRef = useRef<string>("");
+
+  const [isBusy, setIsBusy] = useState<boolean>(false)
+  const [transcript, setTranscript] = useState<string>("");
+
+  const { result, isLoaded, isListening, error, init, start, stop, release } =
+    useCheetah();
+
+  useEffect(() => {
+    if (result !== null) {
+      setTranscript((prev) => {
+        let newTranscript = prev + result.partialTranscript;
+
+        if (result.isComplete) {
+          newTranscript += " ";
```

Comment:
I think a new line instead of a space may look a little better in this case. The line can get quite long.